File: src/main/java/array/Problem_977_sortedSquares.java
package array;

import java.util.Arrays;

public class Problem_977_sortedSquares {

    public int[] sortedSquares(int[] A) {
        int[] A2 = new int[A.length];
        for (int i = 0; i < A.length; i++) {
            A2[i] = (int) Math.pow(A[i],2);
        }
        Arrays.sort(A2);
        return A2;
    }
}

File: src/main/scala/com/strings/leetcode/array/Problem_977_sortedSquares.scala
package com.strings.leetcode.array

object Problem_977_sortedSquares {


  def sortedSquares(A: Array[Int]): Array[Int] = {
    val powered: Array[Int] =  A.map(x => scala.math.pow(x,2).toInt)
    powered.sortWith(_ < _)
  }
}
